src/pybind/mgr/orchestrator/module.py

```diff
@@ -649,10 +652,10 @@ def _daemon_add_misc(self, daemon_type=None, placement=None, inbuf=None):
                 raise OrchestratorValidationError(usage)
             spec = ServiceSpec.from_json(yaml.safe_load(inbuf))
         else:
-            placement = PlacementSpec.from_string(placement)
-            placement.validate()
+            placement = PlacementSpec.from_string(placement)  # type: ignore
+            spec = ServiceSpec(daemon_type, placement=placement)  # type: ignore
```

Comment:
you should not get any mypy errors here. Can you paste the error?

Reply:
add an additional commit to fix the mypy errors (https://github.com/ceph/ceph/pull/34843/commits/f132798b4c1b2f87af4205257daef020f9ab30fb)

Reply:
ah yes, that's a limitation from mypy. you cannot re-assign a variable with a different type:

```python
def func() -> None:
  i = 1
  i = ""    
```

```
$ mypy func.py 
func.py:3: error: Incompatible types in assignment (expression has type "str", variable has 
 type "int")
```

maybe simply use a different name for the placement string?